core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -619,6 +618,29 @@ public PlanOptimizers(
                                 new PruneTableScanColumns(metadata),
                                 new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))));
 
+        // Try to pushdown aggregations before MultipleDistinctAggregationToMarkDistinct.
+        // This needs to be done after redirections.
+        builder.add(
+                new IterativeOptimizer(
+                        metadata,
+                        ruleStats,
+                        statsCalculator,
+                        estimatedExchangesCostCalculator,
+                        ImmutableSet.of(
+                                new PushAggregationIntoTableScan(metadata))),
+                new IterativeOptimizer(
+                        metadata,
+                        ruleStats,
+                        statsCalculator,
+                        estimatedExchangesCostCalculator,
+                        ImmutableSet.<Rule<?>>builder()
+                                .addAll(columnPruningRules)
+                                .addAll(projectionPushdownRules)
```

Comment:
`projectionPushdownRules` doesn't contain `PushDownDereferencesThroughMarkDistinct`. It looks like an omission. Consider adding the rule to the set.

Reply:
https://github.com/trinodb/trino/pull/8962